### PR TITLE
Warning added for save only action

### PR DESCRIPTION
**Pull request by kotewar**
## Description

For save only action, it is important for users to know that the save has failed due to any reason, hence we've added a warning to show the same.

## Motivation and Context

Make user experience better by highlighting any warnings/issues in the workflow run

## How Has This Been Tested?

Local workflow 

Save action - https://github.com/kotewar/test-continue-on-error-for-cache/actions/runs/3742275202
Default cache action - https://github.com/kotewar/test-continue-on-error-for-cache/actions/runs/3742531272/jobs/6353592058#step:13:665

## Screenshots (if appropriate):
NA 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (add or update README or docs)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
 	// Use 'forwardPorts' to make a list of ports inside the container available locally.
 	// "forwardPorts": [],
 	// Use 'postCreateCommand' to run commands after the container is created.
-	"postCreateCommand": "npm install && npm run build"
+	"postCreateCommand": "npm install"
```
Comment (kotewar) [inline .devcontainer/devcontainer.json:9]: npm run builds take a very long time while setting up the codespace, hence removing it from here.

File: dist/save/index.js [hunk 2]
```diff
@@ -41063,33 +41063,34 @@ const utils = __importStar(__webpack_require__(443));
 process.on("uncaughtException", e => utils.logWarning(e.message));
 function saveImpl(stateProvider) {
     return __awaiter(this, void 0, void 0, function* () {
+        let cacheId;
         try {
             if (!utils.isCacheFeatureAvailable()) {
-                return;
+                return 0;
```
Comment (kotewar) [inline dist/save/index.js:41069]: We will return 0 when we don't want to log a warning, and -1 when we want to log a warning.

Comment (bishal-pdMSFT) [inline dist/save/index.js:41069]: 0 may be a real cacheId as DB entry starts at 0. Should we use something like -2

Comment (kotewar) [reply dist/save/index.js:41069]: Yes we can. I was thinking, in the future we can also map the negative numbers to some error types to decide how to log/show error to the user.

Comment (kotewar) [reply dist/save/index.js:41069]: Used -2 now.

Comment (bishal-pdMSFT) [reply dist/save/index.js:41069]: Lets do that now itself. We can define an enum on cache return type.

Comment (bishal-pdMSFT) [reply dist/save/index.js:41069]: Actually re-thinking about this. Why do we need to return id here. Can we return error here. Any error and SaveOnly should print warning message. No?

Comment (kotewar) [reply dist/save/index.js:41069]: Does this mean we also need to implement error handling in save.ts?

**Review by bishal-pdMSFT — APPROVED** [review]

**Pull request merged**